src/gdgem/GdGemBase.cpp

```diff
@@ -429,38 +430,17 @@ void GdGemBase::processingThread() {
 
   TSCTimer ReportTimer;
   unsigned int DataIndex;
-  /* Time for performance measurement
-  Timer duration;
-  duration.now();
-  int cnt = 0;
-  double avg = 0;
-  double total = 0;
-  int rep = 0;
-  */
+
   while (true) {
-    // stats_.fifo_free = InputFifo.free();
     if (!InputFifo.pop(DataIndex)) {
       stats_.ProcessingIdle++;
-      usleep(1);
+      usleep(50); // reduces CPU usage from 100% to ~6% when idle
```

Comment:
don't this also mean that the program/kernel must be able to buffer up 49 additional usecs of data? Is that what the increased Rx/Tx buffer size is for?

Reply:
Yes and no.
The receive() function is basically a system call and it will wait for data for 0.1 second. If 
it hasn' gotten any I allow our thread to usleep. So yes if data arrives just after we exit then
there will be buffering, but this will only happen at low data rates.
The Rx buffers have not been increased, only the Tx buffers, and this was only done to 
ensure a known state. Otherwise it would be OS dependent and I am running on both Linux and Mac where these sizes are different.